ludwig/data/postprocessing.py

```diff
@@ -36,15 +39,35 @@ def postprocess(
         # Only save unprocessed output on the coordinator
         skip_save_unprocessed_output = True
 
-    postprocessed = {}
+    saved_keys = set()
+    if not skip_save_unprocessed_output:
+        _save_as_numpy(predictions, output_directory, saved_keys)
+
     for of_name, output_feature in output_features.items():
-        postprocessed[of_name] = output_feature.postprocess_predictions(
-            predictions[of_name],
+        predictions = output_feature.postprocess_predictions(
+            predictions,
             training_set_metadata[of_name],
             output_directory=output_directory,
-            skip_save_unprocessed_output=skip_save_unprocessed_output
+            backend=backend,
         )
-    return postprocessed
+
+    # Save any new columns but do not save the original columns again
```

Comment:
why do we need this two times? Can't we do it only once at the end?
Will likely also remove the need for `saved_keys`

Reply:
At least in the original implementation, existing columns were written to numpy before postprocessing, while new columns were written after postprocessing. This change is meant to preserve that behavior, but we can definitely remove the earlier lines if that change is acceptable.